src/WebJobs.Script.WebHost/Middleware/HostWarmupMiddleware.cs

```diff
@@ -40,17 +43,59 @@ public async Task Invoke(HttpContext httpContext)
                     PreJitPrepare(WarmUpConstants.LinuxJitTraceFileName);
                 }
 
+                ReadRuntimeAssemblyFiles();
+
                 await WarmUp(httpContext.Request);
             }
 
             await _next.Invoke(httpContext);
         }
 
+        internal void ReadRuntimeAssemblyFiles()
+        {
+            try
+            {
+                string[] allFiles = Directory.GetFiles(_assemblyLocalPath, "*.dll", SearchOption.TopDirectoryOnly);
+                // Read File content in 4K chunks
+                int maxBuffer = 4 * 1024;
+                byte[] chunk = new byte[maxBuffer];
+                foreach (string file in allFiles)
+                {
+                    // Read file content to avoid disk reads during specialization. This is only to page-in bytes.
+                    ReadFileInChunks(file, chunk, maxBuffer);
+                }
+                _logger.LogDebug(new EventId(100, nameof(ReadRuntimeAssemblyFiles)), "Number of files read: {allFilesCount}.", allFiles.Count());
+            }
+            catch (Exception ex)
+            {
+                _logger.LogDebug(new EventId(100, nameof(ReadRuntimeAssemblyFiles)), ex, "Reading ReadRuntimeAssemblyFiles failed");
+            }
+        }
+
+        private void ReadFileInChunks(string file, byte[] chunk, int maxBuffer)
+        {
+            try
+            {
+                using (FileStream fileStream = new FileStream(file, FileMode.Open, FileAccess.Read))
+                {
+                    while (fileStream.Read(chunk, 0, maxBuffer) != 0)
+                    {
+                        // Read one byte for every 4K bytes - 4K is default OS page size. This will help avoid disk read during specialization
+                        var randomByte = Convert.ToInt32(chunk[0]);
```

Comment:
@safihamid / @brettsam - updated PR to read 1 byte for every 4K

Reply:
Can you please read a random byte in that 4K array instead of element 0? This is to make sure OS paging doesn't outsmart byte 0.

Reply:
done!